File: 백준/Silver/2004. 조합 0의 개수/조합 0의 개수.py
import sys
def print(*args, sep=" ", end="\n"): return sys.stdout.write(sep.join(map(str, args)) + end)
def input(): return sys.stdin.readline().rstrip()
#MOD = 1_000_000_007

def main():
    n, m = map(int , input().split())
    def cp(n, m):
        cnt = 0
        while n >= m:
            n //= m
            cnt += n
        return cnt
    cnt2 = cp(n, 2)-cp(m, 2)-cp(n-m, 2)
    cnt5 = cp(n, 5)-cp(m, 5)-cp(n-m, 5)
    print(min(cnt2, cnt5))
    return 0
if __name__ == '__main__':
    sys.exit(main())

File: 백준/Silver/2004. 조합 0의 개수/조합 0의 개수.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int cp(ll n, ll m){
    int cnt = 0;
    while(n >= m){
        n /= m;
        cnt += n;
    }
    return cnt;
}

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    ll n, m; cin >> n >> m;
    int cnt2 = cp(n, 2)-cp(m, 2)-cp(n-m, 2);
    int cnt5 = cp(n, 5)-cp(m, 5)-cp(n-m, 5);
    cout << min(cnt2, cnt5);
    return 0;
}